### PR TITLE
CATTY-378 Implemented AskBrick

### PR DESCRIPTION
**Pull request by d4n3x**
Implemented the AskBrick including some tests in swift.
This Brick prompts a user defined question. The answer is saved to a user variable.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by aeluna — CHANGES_REQUESTED** [review]
Thank you for the PR!
Please also include the new brick in the file ValidProjectAllBricks0992.xml.

**Review by aeluna — APPROVED** [review]

File: src/Catty/AskBrick+Instruction.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        guard let question = self.question else { fatalError("Unexpected found nil.") }
+        guard let userVariable = self.userVariable else { fatalError("Unexpected found nil.") }
+
+        return CBInstruction.highPriorityExecClosure { context, _, _ in
+            Util.askUser(forTextAndPerformAction: #selector(self.callbackSubmit),
+                         target: self,
+                         cancelAction: #selector(self.callbackCancel),
+                         with: userVariable,
+                         promptTitle: nil,
+                         promptMessage: question.getDisplayString(),
```
Comment (m-herold) [inline src/Catty/AskBrick+Instruction.swift:35]: Please remove the quotes from the alert message:

Screenshot 2020-09-29 at 07 10 13

**Review by m-herold — CHANGES_REQUESTED** [review]
Thanks, looks good! I have added a few suggestions directly to the code. Here is one general problem:

When running for example this project (https://share.catrob.at/app/project/c91ddd8a-cb43-11ea-a5c3-005056a36f47) I have to click on the screen before the project continues after submitting an answer.

**Review by m-herold — CHANGES_REQUESTED** [review]
One more thing: please try to remove the "Cancel" button at the alert to be compliant with Catroid.

File: src/CattyTests/Bricks/AskBrickTests.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        brick.script = script
+
+        let logger = CBLogger(name: "Logger")
+        broadcastHandler = CBBroadcastHandler(logger: logger)
+        let formulaInterpreter = FormulaManager(stageSize: Util.screenSize(true), landscapeMode: false)
+        scheduler = CBScheduler(logger: logger, broadcastHandler: broadcastHandler, formulaInterpreter: formulaInterpreter, audioEngine: AudioEngineMock())
+        context = CBScriptContext(script: script, spriteNode: spriteNode, formulaInterpreter: formulaInterpreter, touchManager: formulaInterpreter.touchManager)
+    }
+
+    func testAnswerNormal() {
```
Comment (m-herold) [inline src/CattyTests/Bricks/AskBrickTests.swift:71]: Please provide another test case which documents that the value is not changing when submitting "" (no input).

**Review by m-herold — CHANGES_REQUESTED** [review]

**Review by m-herold — APPROVED** [review]

**Comment by m-herold** [comment]
Thanks @d4n3x !

**Pull request merged**